granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -28,21 +28,19 @@
 }
 
 
-def get_cgroups(pid: int) -> List[Tuple[str, List[str], str]]:
+def get_cgroups(process: Optional[psutil.Process] = None) -> List[Tuple[str, List[str], str]]:
```

Comment:
You broke the API, please next time do `Union[int, Process]`

Reply:
Fixed, not broke. You need a Process object in this implementation for proper error handling

Reply:
Did you read the PR description?

Reply:
What would you rely on? Runtime failures? Documentation? 

Runtime failures would happen anyway, so the reliance is not strictly on static type checking. And you can't rely on documentation because it needs to be read by people. You can't rely on versioning either, because it doesn't tell you what and where to fix.

Reply:
The only alternative would be to leave everything broken and create a new API, which defeats the purpose of the change.

Reply:
I agree with @d3dave here. We rely on mypy here. This is an improvement over the previous api.